File: tests/go/functions/function_early_return.go
// vybe-test: go/functions/function_early_return
// origin: languages/go/tests/go/test_functions.rs

package main
import "fmt"
func check(n int) string { if n < 0 { return "neg" }
return "ok" } var __buf string

// __p appends one line, __pr appends without a newline.
func __p(s string) { __buf = __buf + s + "\n" }

func __pr(s string) { __buf = __buf + s }

// __check ends the program unless the collected output equals want. The final
// Println contributes a trailing newline the expected line vector never
// carried, so both forms are accepted.
func __check(want string) {
	if __buf != want && __buf != want+"\n" {
		fmt.Println("FAIL: want [" + want + "] got [" + __buf + "]")
		panic("assertion failed")
	}
}

func main() { __p(fmt.Sprint(check(-1)))
__p(fmt.Sprint(check(1)))
__check("neg\nok")
}

File: tests/lua/functions/function_early_return.lua
-- vybe-test: lua/functions/function_early_return
-- origin: languages/lua/tests/lua/test_functions.rs

local __w1 = "-1"
local __i = 0

function sign(n)
  if n < 0 then return -1 end
  if n > 0 then return 1 end
  return 0
end
do local __t = tostring(sign(-3)); __i = __i + 1
  if __i == 1 and __t ~= __w1 then error("FAIL: want [" .. __w1 .. "] got [" .. __t .. "]") end end

if __i == 0 then error("FAIL: no output, wanted [" .. __w1 .. "]") end
